src/db/util/raw-model.ts

```diff
@@ -73,6 +75,14 @@ export type DestroyFn<F extends FieldDefinition> = (args: {
 
 export type TruncateFn = (trx?: Knex.Transaction<any, any>) => Promise<void>;
 
+export type CountFn<F extends FieldDefinition, AdditionalArgs = {}> = (
+  args?: {
+    where?: WhereCond<F>;
+    trx?: Knex.Transaction<any, any>;
+    distinct?: Array<keyof InstanceDataOf<F>>;
```

Comment:
Like I said in the other comment, I think we should document that `COUNT` filters out `NOT NULL` values if expression is provided instead of `*`
```suggestion
    /**
     * Will produce `SELECT COUNT(DISTINCT "columnName")` and if there
     * is an expression in `COUNT`, it will compute the number of input
     * rows in which the input value is **not null**. So, besides selecting
     * unique values of "columnName", it will also filter out `NULL` values
     */
    distinct?: Array<keyof InstanceDataOf<F>>;
```